### PR TITLE
Support compilation on Windows (tested on Visual Studio 2015)

### LINKED ISSUE #145: libucl 0.8.0 release is pretty broken on Windows 10 + Visual Studio 2015 Community edition
Using CMake am able to build the libucl.sln file for 0.8.0 but actually building the source seems to be out of the question. 100s of errors and warnings. Is Windows a supported platform. Have there been any changes to upgrade this library to build cleanly on Windows since 0.8.0?

### PR DESCRIPTION
**Pull request by odiszapc**
Fix #145

### CODE REVIEW

**Comment by odiszapc** [comment]
@vstakhov Did you have a chance to look at this PR?

**Comment by dxramax** [comment]
I have tried to compile libucl using Microsoft Visual Studio 2017 Community Edition with no luck, I applied the changes in this PR and it worked like a charm, thanks a lot.

File: src/ucl_util.c [hunk 1]
```diff
@@ -40,9 +40,56 @@
...
+    char val[512]; // TODO : use appropriate constants macro
+    _splitpath(
+        path,
+        drive, // drive
+        dir,   // dirname
+        NULL, // doesn't use filename
+        NULL  // doesn't use ext
+    );
+    sprintf(val, "%s%s", drive, dir);
+    return strdup (val);
```
Comment (vstakhov) [inline src/ucl_util.c]: Memory leak here.

File: src/ucl_util.c [hunk 2]
```diff
@@ -40,9 +40,56 @@
...
+    char dir[512]; // TODO : use appropriate constants macro
+    char val[512]; // TODO : use appropriate constants macro
+    _splitpath(
+        path,
+        drive, // drive
+        dir,   // dirname
+        NULL, // doesn't use filename
+        NULL  // doesn't use ext
+    );
+    sprintf(val, "%s%s", drive, dir);
```
Comment (vstakhov) [inline src/ucl_util.c]: Potential out-of-boundary write (never use `sprintf`).

File: src/ucl_util.c [hunk 3]
```diff
@@ -40,9 +40,56 @@
...
 #include <libgen.h> /* For dirname */
+#else
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+char *dirname(char *path)
+{
+    char drive[3];
+    char dir[512]; // TODO : use appropriate constants macro
+    char val[512]; // TODO : use appropriate constants macro
```
Comment (vstakhov) [inline src/ucl_util.c]: Why not use `PATH_MAX` for dirname? For filename there is a dedicated constant in Unix.

File: src/ucl_util.c [hunk 4]
```diff
@@ -40,9 +40,56 @@
...
+    _splitpath(
```
Comment (vstakhov) [inline src/ucl_util.c]: Why not use `_splitpath_s`?

**Pull request closed without merging**